common/state/state_kv.cpp

```diff
@@ -71,13 +71,19 @@ class pstate::data_node {
             return block_num_;
         }
 
-        ByteArray serialize_data() {
+        ByteArray& serialize_data() {
             ByteArray header = make_offset(block_num_, free_bytes_);
             std::copy(header.begin(), header.end(), data_.begin());
             return data_;
         }
 
```

Comment:
do you really want to return a reference to a data structure inside the object? at a minimum, can we make the result a const ByteArray&?